api/types/types.go

```diff
@@ -151,9 +151,12 @@ type CopyConfig struct {
 // ContainerPathStat is used to encode the header from
 // GET "/containers/{name:.*}/archive"
 // "Name" is the file or directory name.
+// If path is a dir, dsize means the size of path and
+// his children. Otherwise, DSize is 0.
 type ContainerPathStat struct {
 	Name       string      `json:"name"`
 	Size       int64       `json:"size"`
+	DSize      int64       `json:"dsize"`
```

Comment:
The dsize works when the path is a directory. size only report the dir size, but We want to get the Whole size of the files in this dir. Then I introduce a DSize here.